terminate/onpledge.go

```diff
@@ -0,0 +1,109 @@
+package terminate
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	filbig "github.com/filecoin-project/go-state-types/big"
+	"github.com/filecoin-project/go-state-types/builtin"
+	"github.com/filecoin-project/go-state-types/builtin/v13/miner"
+	lotusapi "github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types"
+	miner8 "github.com/filecoin-project/specs-actors/v8/actors/builtin/miner"
+	"github.com/glifio/go-pools/util"
+)
+
+func TermPenaltyOnPledge(
+	ctx context.Context,
+	api *lotusapi.FullNodeStruct,
+	ts *types.TipSet,
+	filToPledge *big.Int,
+	sectorSize uint64,
+	ratioVerified float64,
+) (cost *big.Int, penalty *big.Int, err error) {
+	smoothedPow, err := util.TotalPowerSmoothed(ctx, api, ts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	smoothedRew, err := util.ThisEpochRewardsSmoothed(ctx, api, ts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	height := ts.Height()
+	activation := 4071894
+	expiration := 5619449
```

Comment:
Should these still be hardcoded?

Reply:
I change the API so there are passed in now.